azure-pipelines.yml

```diff
@@ -43,127 +43,45 @@ stages:
-            tasks: clean test -DTEST_SUBSCRIPTION_KEY=$(hmi-apim-sub-key) -DTEST_URL=https://hmi-apim-svc-sbox.azure-api.net -DTEST_HOST=hmi-apim-svc-sbox.azure-api.net
-            publishJUnitResults: true
-            testResultsFiles: '**/TEST-*.xml'
-            javaHomeOption: 'JDKVersion'
-            jdkVersionOption: '1.11'
-            sonarQubeRunAnalysis: false
 
-  - stage: TESTS2
+  - stage: TESTS3
     dependsOn: BUILD
-    displayName: Tests - Security
+    displayName: Tests - Unit
     jobs:
-      - job: testsec
-        pool:
-          vmImage: 'ubuntu-latest'
-        steps:
-        - task: AzureCLI@2
-          displayName: Build and run API ZAP scan
-          inputs:
-            azureSubscription: $(subscription)
-            scriptType: bash
-            scriptLocation: inlineScript
-            inlineScript: |
-              echo "Obtaining container key..."
-              KEY=$(az storage account keys list -g $(devopsResourceGroup) --account-name $(devopsStorageAccount) --query "[0].value" -o tsv)
-              echo $KEY
-              echo "Creating ACI ZAP container..."
-              az container create -g $(devopsResourceGroup) -n hmi-aci-zap-sbox --image $(owaspZapImage) --azure-file-volume-account-name $(devopsStorageAccount) --azure-file-volume-account-key $KEY --azure-file-volume-share-name $(owaspZapVolume) --azure-file-volume-mount-path /zap/wrk/ --command-line "zap-baseline.py -t https://hmi-apim-svc-sbox.azure-api.net -d -l FAIL -x api_scan.xml -r api_scan.html -w api_scan.md"
-              sleep 30
-              echo "Downloading ZAP report..."
-              az storage file download --account-name $(devopsStorageAccount) --account-key $KEY -s $(owaspZapVolume) -p api_scan.html --dest $(System.DefaultWorkingDirectory)/api_scan.html
-
-        - task: AzureCLI@2
-          displayName: Pulish test results
-          condition: succeededOrFailed()
-          inputs:
-            azureSubscription: $(subscription)
-            scriptType: bash
-            scriptLocation: inlineScript
-            inlineScript: |
-              reportFile=$(System.DefaultWorkingDirectory)/api_scan.html
-              mdReportFile=$$(System.DefaultWorkingDirectory)/api_scan.md
-              echo "##vso[task.uploadfile]$reportFile"
-              # echo "##vso[task.uploadsummary]$mdReportFile"
-
-        - task: AzureCLI@2
-          displayName: Delete ACI container
-          condition: succeededOrFailed()
-          inputs:
-            azureSubscription: $(subscription)
-            scriptType: bash
-            scriptLocation: inlineScript
-            inlineScript: |
-              az container delete -g $(devopsResourceGroup) -n hmi-aci-zap-sbox --yes
\ No newline at end of file
+    - job: testunit
+      variables:
+        hmi-apim-sub-key: $[stageDependencies.BUILD.Terraform.outputs['exportKeyVaultSecret.hmi-apim-sub-key']]
+        hmi-apim-host: hmi-apim-svc-sbox.azure-api.net
```

Comment:
Does the above variable get overwritten when the build runs for test environment?